### ISSUE TITLE
Emulator support?/and or bug during emulation.

### ISSUE THREAD

**Issue by Tiki-pixel** [GearsProgress/Poke_Transporter_GB#1]
Using mGBA 0.10.3 as the emulator, Poke Transporter GB works fine up until the trading part. To initiate the .GBA rom, I replaced it with a Generation 3 rom using the "Replace Rom" option, and the multiboot works fine. Then I created a multiplayer window (or boot up another window of mGBA) with the generation 1 or 2 game. In this test, I used Emerald, crystal and silver (Trying out two Gen 2 games with Emerald).

When preparing to trade, Pokemon Crystal never detected anyone on the link cable, and a message of "Your friend is not ready" played. It was a shame but to-be expected. But something caught my eye. I tried the mGBA nightly build (build 2017 July 8th and 9), and was surprised that the link cable trade connected... or it did for a split second. As soon as the dialogue ended, the crystal and silver game froze immediately for an indefinite amount of time. But it did indeed seem to communicate.

So this brings the question of "Will there be emulation support?", or was emulation support planned but never built? Reason being is that I personally really love this method of transfering Generation 1-2 to Generation 3 games, but I just don't have the hardware. Also I don't want to modify my pokemon just to use a save editor to plop them into my save. Either way, thank you GearsProgress for making this wonderful rom!

**Comment by GearsProgress** [comment]
After talking with a member of the mGBA development team, it sounds like they're currently working on rewriting the link cable system to eventually include net play. The good news is that once that is done, implementing the different link cables shouldn't be too hard. The unfortunate news is that it might be a while until mGBA reaches that point. But I'll keep an eye on it and look into some other solutions to get Poke Transporter GB working in mGBA in the meantime!

**Comment by AndHesLoose92** [comment]
Any word on emulation capabilities? I just caught all of the original 251 Pokemon and would love the ability to get them into later games.

**Comment by GearsProgress** [comment]
Unfortunately at the moment, no. Currently the priority is fixing a few major bugs and adding language compatibility- but after that I'm probably going to do some more research on emulation possibilities. Sorry for the wait!! 😅

**Comment by JeodC** [comment]
This sort of application would be immensely useful as a standalone thing for emulation handhelds. In theory it would need to be able to read `.sav` and `.srm` files, find and lift the information from them, *transfer* the information to another eligible `.sav` or `.srm` file, and modify the original to remove the transferred data. All of that underneath the wonderful authentic-looking UI would be a huge hit.

To further break down steps on a headless (no display manager) system:
```
1 - Load `.sav` or `.srm` files via list populated from a given config file path (cat | grep retroarch.cfg maybe)
2 - Of files that are found, determine which are eligible (attempt to identify/read them)
2.5 - Of identified/eligible files, check which have available Pokemon for transfer (maybe check boxes only)
3 - Present eligible list to user to select the host save file
4 - Repeat process for client save file
5 - Sanity check - ensure host and client save files are both eligible for use, host has available Pokemon for transfer, client has available Box space
6 - Ask user to select which Pokemon to transfer
7 - Perform transfer - transfer from host to client, verify success by rereading data on client maybe
8 - On success, wipe data from host
9 - Finalize both save file rewrites
10 - Notify user of success
```

Of course PKHex and the like can be used, but it's always preferable to have something confined to the device in use and prevent the need to use a computer.

**Comment by hybrid836** [comment]
Hi,

Just finished a full Pokedex run of Crystal and wanted to transfer them to Emerald. I understand that (at least) mGBA detects what type of ROM is playing (.gba, .gb, etc.) and sets the link cable type based on that. Since PokeTransporter GB and the destination game are .gba ROMs, any call to the link cable protocol is going to use the GBA link protocol and not the GB one. I wonder if it would be just as simple as changing some of the code in mGBA (seems like MultiplayerController.cpp is the file that takes care of link cable stuff) to force it to load the GB protocol when certain ROMs are loaded. If the link cable code on the mGBA side are going to be rewritten anyway, it might be better just to wait for that implementation to roll out.

<img width="1333" height="520" alt="Image" src="https://github.com/user-attachments/assets/aac453ac-1641-422f-904d-15a1b97046e5" />
Here's a screenshot of my mGBA instances trying to connect. The instance running PT GB is stuck on this screen endlessly waiting to connect :( Is there supposed to be a way to cancel out out of the connection attempt?

Specs:
mGBA version: 0.10.4
PokeTransporter GB version: 1.1.3

**Comment by GearsProgress** [comment]
I took a look into the code and, unfortunately it seems to be a bit more complicated than expected. It looks like the cable is tied directly to the current core, which in turn is specified by the current running game. I also looked into a few other emulators with link cable support (such as VBA) and I didn't have success with those either. The search for the white whale continues 🙃

**Comment by Tiki-pixel** [comment]
It's been a while, but I have a crackhead idea that may possibly work? Of course theres a chance that it wont. After reading the occasional updates via email, I wanted to try to chip in.

I remember the idea of a "middle-man" program, so what if there was exactly that? A possible "emu only" mode. From what im understanding is that communication from GB and GBA games just straight up doesn't work. But the same format game does, ex; GB-GB, GBA-GBA, GBC-GBC

What if there was a version or just an application thats technically a Gameboy version (GB rom/GBC rom) so it can communicate with gen 1-2 parent game? Then via a trade, the Gameboy version of the rom then can save the transfered pokemon, and then communicate to the original GBA transporter?

This probably wont work because of the issue with the link cable, but may because of instead limiting on how Pokémon files are saved, you can possibly use custom save information to then transfer?

Tl;dr creating a middle man program/rom thats the same type of the gen 1-2 game, trading between the two, and then the middle man program communicates to the original game.

Im loving the progress and I truthfully think that once emulator support has been found/exploited, it'll be extremely useful.

Thank you GearsProgress!

**Comment by hybrid836** [comment]
@GearsProgress Yeah, that's what I figured. If it was that easy, someone would've done it already lol. Thanks for taking a look. I've got an EZ-Flash coming in soon, so I'll test out your ROM on real hardware.

**Comment by theshinyumbreon258** [comment]
@GearsProgress isnt there an emulator that can connect to dolphin? what if instead of having the poketransportergb be on the gba, we have it on the gamecube? that way we can connect two instances of a gba to the gc emu, with the two different games loaded on the gba emus  and the gc emu handles the transfer? we might have to drop the dream dex function unless we can create a new save on a memcard that the gc emu wont flag as corupted data

**Comment by theshinyumbreon258** [comment]
i think my idea would be more feasible, and safer if we used a GameCube as a middleman of sorts, that way we don't have to deal with different voltage levels or modifying an existing emulator to make it work
